uvicorn/supervisors/multiprocess.py

```diff
@@ -23,11 +23,32 @@ def __init__(self, config, target, sockets):
         self.processes = []
         self.should_exit = threading.Event()
         self.pid = os.getpid()
+        self.child_pids = []
```

Comment:
Getting back to https://github.com/encode/uvicorn/pull/522#discussion_r406355579 …

I haven't tried the Docker test yet, but I'm not sure I understand _why_ using a separate attribute to store PIDs of `self.processes` is necessary for this to work.

In my mind, `child_pids` is equivalent to `[p.pid for p in self.processes]`. Isn't that guaranteed/can the `.pid` of a process object change in its lifespan?

Reply:
while I agree for the multiprocess part,  I fail to see how to tackle this on the reload part

Reply:
The reload part runs a single process - and it seems to be stored on `self.process`. We don't even need a `for` loop, only to `stop_subprocess(self.process.pid)`, correct? Or am I missing something?

Reply:
you're right indeed, dunno why I've been stuck with that in mind, just had to setup the same `self.process = []` for the basereload so that the `test_statreload` and `test_watchdogreload` pass
it would fail otherwise since self.process would be non existant